tests/test_preprints.py

```diff
@@ -462,13 +467,114 @@ def test_format_preprint(self):
 
         assert nodes == {}
 
+    def test_format_thesis(self):
```

Comment:
Driveby comment: There is a lot going on in this test. Might consider adding comments explaining what each sub section is testing. E.g., # tests that graph node contains correct thesis information